lib/utils/cli.go

```diff
@@ -211,40 +211,43 @@ func formatErrorWriter(err error, w io.Writer) {
 }
 
 func formatCertError(err error) string {
+	const unknownAuthority = `WARNING:
+The proxy you are connecting to has presented a certificate signed by a
+unknown authority. This is most likely due to either being presented
+with a self-signed certificate or the certificate was truly signed by an
+authority not known to the client.
+
+If you know the certificate is self-signed and would like to ignore this
+error use the --insecure flag.
+If you have your own certificate authority that you would like to use to
+validate the certificate chain presented by the proxy, set the
+SSL_CERT_FILE and SSL_CERT_DIR environment variables respectively and try
+again.
+
+If you think something malicious may be occurring, contact your Teleport
+system administrator to resolve this issue.
+  `
+
 	switch innerError := trace.Unwrap(err).(type) {
 	case x509.HostnameError:
 		return fmt.Sprintf("Cannot establish https connection to %s:\n%s\n%s\n",
 			innerError.Host,
 			innerError.Error(),
 			"try a different hostname for --proxy or specify --insecure flag if you know what you're doing.")
 	case x509.UnknownAuthorityError:
-		return `WARNING:
-
-  The proxy you are connecting to has presented a certificate signed by a
-  unknown authority. This is most likely due to either being presented
-  with a self-signed certificate or the certificate was truly signed by an
-  authority not known to the client.
-
-  If you know the certificate is self-signed and would like to ignore this
-  error use the --insecure flag.
-
-  If you have your own certificate authority that you would like to use to
-  validate the certificate chain presented by the proxy, set the
-  SSL_CERT_FILE and SSL_CERT_DIR environment variables respectively and try
-  again.
-
-  If you think something malicious may be occurring, contact your Teleport
-  system administrator to resolve this issue.
-`
+		return unknownAuthority
 	case x509.CertificateInvalidError:
 		return fmt.Sprintf(`WARNING:
-
   The certificate presented by the proxy is invalid: %v.
-
   Contact your Teleport system administrator to resolve this issue.`, innerError)
-	default:
-		return ""
 	}
+
+	// Check for less explicit errors
+	if strings.Contains(err.Error(), "certificate is not trusted") {
+		return unknownAuthority
+	}
```

Comment:
nit: are we doing the string comparison because it's an aggregate? If so, is it possible to get individual errors out of it?

Reply:
Unfortunately, my original plan was to use `errors.Is/errors.As` here, as I made changes to the Aggregate error type to properly support these, however, I was foiled by the Go stdlib. The x509 package makes generous use of `errors.New` rather than better error types.